src/engine/Union.cpp

```diff
@@ -177,43 +184,40 @@ ProtoResult Union::computeResult([[maybe_unused]] bool requestLaziness) {
           Result::getMergedLocalVocab(*subRes1, *subRes2)};
 }
 
-void Union::computeUnion(
-    IdTable* resPtr, const IdTable& left, const IdTable& right,
-    const std::vector<std::array<size_t, 2>>& columnOrigins) {
-  IdTable& res = *resPtr;
-  res.resize(left.size() + right.size());
-
-  static constexpr size_t chunkSize = 1'000'000;
+// _____________________________________________________________________________
+void Union::copyChunked(auto beg, auto end, auto target) const {
+  size_t total = end - beg;
+  for (size_t i = 0; i < total; i += chunkSize) {
+    checkCancellation();
+    size_t actualEnd = std::min(i + chunkSize, total);
+    std::copy(beg + i, beg + actualEnd, target + i);
+  }
+}
```

Comment:
Can you make those generic and also use them for `BIND` ?
Either completely generic (then you have to pass in the `checkCancellation()` function manually, so
`ad_utility::copyChunked(beg, end, target, chunkSize, std::bind_front(&checkCancellation, this)`,
Or by making those members of the `Operation` base class.

Reply:
Good point, but I'd prefer to do this in a follow-up PR. Then I can add extensive unit tests and stuff without bloating the scope of this PR.